spring-cloud-stream-reactive/src/main/java/org/springframework/cloud/stream/reactive/StreamEmitterAnnotationBeanPostProcessor.java

```diff
@@ -0,0 +1,279 @@
+							int outputAnnotationCount = StreamAnnotationCommonMethodUtils.outputAnnotationCount(item);
+							validateStreamEmitterMethod(item, outputAnnotationCount, methodAnnotatedOutboundName);
+							invokeSetupMethodOnToTargetChannel(item, k, methodAnnotatedOutboundName);
+						}
+				));
+				this.running = true;
+			}
+		}
+		finally {
+			lock.unlock();
+		}
+	}
+
+	@SuppressWarnings({"rawtypes", "unchecked"})
+	private void invokeSetupMethodOnToTargetChannel(Method method, Object bean, String outboundName) {
+		Object[] arguments = new Object[method.getParameterTypes().length];
+		Object targetBean = null;
+		for (int parameterIndex = 0; parameterIndex < arguments.length; parameterIndex++) {
+			MethodParameter methodParameter = new SynthesizingMethodParameter(method, parameterIndex);
+			Class<?> parameterType = methodParameter.getParameterType();
+			Object targetReferenceValue = null;
+			if (methodParameter.hasParameterAnnotation(Output.class)) {
+				targetReferenceValue = AnnotationUtils.getValue(methodParameter.getParameterAnnotation(Output.class));
+			}
+			else if (arguments.length == 1 && StringUtils.hasText(outboundName)) {
+				targetReferenceValue = outboundName;
+			}
+			if (targetReferenceValue != null) {
+				targetBean = this.applicationContext.getBean((String) targetReferenceValue);
+				for (StreamListenerParameterAdapter<?, Object> streamListenerParameterAdapter : this.streamListenerParameterAdapters) {
+					if (streamListenerParameterAdapter.supports(targetBean.getClass(), methodParameter)) {
+						arguments[parameterIndex] = streamListenerParameterAdapter.adapt(targetBean,
+								methodParameter);
+						if (FluxSender.class.isAssignableFrom(arguments[parameterIndex].getClass())) {
+							fluxSenders.add((FluxSender) arguments[parameterIndex]);
+						}
+						break;
+					}
+				}
+				Assert.notNull(arguments[parameterIndex], "Cannot convert argument " + parameterIndex + " of " + method
+						+ "from " + targetBean.getClass() + " to " + parameterType);
+			}
+			else {
+				throw new IllegalStateException(StreamEmitterErrorMessages.ATLEAST_ONE_OUTPUT);
+			}
+		}
+		Object result;
+		try {
+			result = method.invoke(bean, arguments);
+		}
+		catch (Exception e) {
+			throw new BeanInitializationException("Cannot setup StreamEmitter for " + method, e);
+		}
+
+		if (!Void.TYPE.equals(method.getReturnType())) {
+			if (targetBean == null) {
+				targetBean = this.applicationContext.getBean(outboundName);
+			}
+			boolean streamListenerResultAdapterFound = false;
+			for (StreamListenerResultAdapter streamListenerResultAdapter : this.streamListenerResultAdapters) {
+				if (streamListenerResultAdapter.supports(result.getClass(), targetBean.getClass())) {
+					Object adapted = streamListenerResultAdapter.adapt(result, targetBean);
+					if (adapted != null && Disposable.class.isAssignableFrom(adapted.getClass())) {
```

Comment:
According to the latest changes it looks like we don't have `Disposable` returns any more.
So, can be dropped altogether here as well.

Also I still don't understand why do you use `class.isAssignableFrom()` instead of `instanceof`.
You have already `Object adapted` in place. What's wrong? Thanks